compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4380,3 +4381,58 @@ declare_lint! {
     "effective visibility of a type is larger than the area in which it can be named",
     @feature_gate = sym::type_privacy_lints;
 }
+
+declare_lint! {
+    /// The `coinductive_overlap_in_coherence` lint detects impls which are currently
+    /// considered not overlapping, but may be considered to overlap if support for
+    /// coinduction is added to the trait solver.
+    ///
+    /// ### Example
```

Comment:
I'm not sure if it's better to a simple but more contrived example here like:
```rust
trait CyclicTrait {}
impl<T: CyclicTrait> CyclicTrait for T {}

trait Trait {}
impl<T: CyclicTrait> Trait for T {}
impl Trait for u8 {} // conflicting impl with the above
```

It took me a while to understand the current example. On the other hand it is reduced from a real-world code, so :man_shrugging:

Reply:
I think the simpler example is fine here too 👍